lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -87,18 +86,21 @@ def __init__(self, config: PbenchServerConfig, logger: Logger, *schemas: ApiSche
 
         api_name = self.__class__.__name__
 
-        if not schemas:
-            raise MissingDatasetNameParameter(api_name, "no schema provided")
-        dset = self.schemas.get_param_by_type(
-            API_METHOD.POST,
+        if len(schemas) != 1:
+            raise MissingDatasetNameParameter(
+                api_name, f"exactly one schema required: found {len(schemas)}"
+            )
+
+        self.schema = schemas[0]
```

Comment:
I don't think we should store this as an instance property; it's not (and probably shouldn't be) used outside the constructor for basic consistency checking. I think just a local `schema` would be better.

Reply:
Indeed; I was skimming too quickly this morning and failed to notice that there was a gap between the diffs -- it's outside the constructor.

Since we do know there's only a single `Schema` it could still use `self.schemas[0]`; but let's just keep the `self.schema`. That seems a bit cleaner outside the constructor.

Reply:
Yeah, Dave.So only I thought of using this `self.schema.` Then I'll leave this code as it is :smile: